docs/contributing.md

```diff
@@ -0,0 +1,121 @@
+of these platforms, however, MacOS packages built on Windows and on Linux will not have a proper 
+icon, as generation of MacOS icons is a non-trivial task on these platforms.
+
+<br/>
+
+### Code Style Guide
+Please note that you should not use a code auto-formatter in this codebase. Please read the following
+documents to learn more about reasons behind this decision and the recommended code style guide. 
+Be sure to carefully read the documents before contributing to this repository:
+- [Rust style guide 1](https://github.com/luna/basegl/blob/master/docs/style-guide.md)
+- [Rust style guide 2](https://github.com/luna/enso/blob/master/doc/rust-style-guide.md) 
+
+<br/>
+
+### Setup
+- **The Rust Toolchain**  
+  This project uses several features available only in the nightly Rust toolchain.
+Please use the [the Rust toolchain installer](https://rustup.rs) to install it:
+
+  ```bash
+  rustup toolchain install nightly-2019-11-04 # Install the nightly channel.
+  rustup default nightly                      # Set it as the default one.
+  rustup component add clippy                 # Install the linter.
+  ```
+
+- **Node Version Manager**  
+  In order to build the web and desktop applications you will need [node](https://nodejs.org) and 
+[npm](https://www.npmjs.com). Please note that the versions available in your system package manager
+will probably crash while building native-extensions. The only known stable solution is to use the 
+[Node Version Manager](https://github.com/nvm-sh/nvm). Please note that installing it from any 
+package manager is officially not supported and can cause issues. Follow the 
+[official guide](https://github.com/nvm-sh/nvm#installing-and-updating) to install it:
+
+  ```bash
+  curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.2/install.sh | bash
+  ```
+
+<br/>
+<br/>
+<br/>
+
+## Working with sources
+As this is a multi-part project with many complex dependencies, it was equipped with a build script
+which both validates your working environment as well as takes care of providing most suitable 
+compilation flags for a particular development stage. In order to learn more about the commands and 
+available options, simply run `./run` and read the manual. The most common options are presented 
+below:
+
+- **Interactive builds**  
+  Run `./run watch` to start a local web-server and a source-file watch utility which will build the 
+  project on every change. Open `http://localhost:8080` to run the application, or 
+  `http://localhost:8080/debug` to see example demo scenes. Please remember to disable the cache in 
+  your browser during development! By default, the script disables heavyweight optimizations to 
+  provide interactive development experience. The scripts are thin wrappers for 
+  [wasm-pack](https://github.com/rustwasm/wasm-pack) and accept the same 
+  [command line arguments](https://rustwasm.github.io/wasm-pack/book/commands/build.html).
```

Comment:
I don't think it is true anymore?
Or please show example of passing arguments through the new wrappers.